devtools/maven/src/main/java/io/quarkus/maven/TrackConfigChangesMojo.java

```diff
@@ -177,6 +168,19 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
         }
     }
 
+    private Path resolvePreviousBuildConfigDump(LaunchMode launchMode) {
+        final Path previousBuildConfigDump = this.recordedBuildConfigFile == null ? null
+                : Path.of(this.recordedBuildConfigFile);
+        if (previousBuildConfigDump == null) {
+            return recordedBuildConfigDirectory.toPath()
+                    .resolve("quarkus-" + launchMode.getDefaultProfile() + "-config-dump");
```

Comment:
I was having a look at the original issue and is it expected we are not taking into account  `quarkus.config-tracking.file-suffix`?

Reply:
We could. We'd need to refactor the current implementation further to do that.
It will come at a cost of always resolving the the build time config and bootstrap the app even on the first build (when the config dump from the previous build does not exist, since we will not know about that).

Reply:
Does it make sense to have a config property for it if we can't honor it?

Reply:
No, it's used but by the process dumping the used config options during the build. This mojo is another client that reads that file. So all the options are actually used.
I think you had a good point though, we could potentially re-use the same option used by the build process in this mojo.
Perhaps we could keep the existing one but use it only if a user configured values explicitly.

Reply:
> No, it's used but by the process dumping the used config options during the build.

Yeah, I had that in mind but does it make sense to make it configurable if it's hard to propagate this config everywhere?

What's the value of allowing people to adjust the suffix?

Reply:
Exactly, when using the [quarkus-build-caching-extension](https://github.com/gradle/quarkus-build-caching-extension/) to make the Quarkus `build` goal cacheable, a recommendation is to check in the dump config in the Git repository.
You may have several versions of this dump config depending on several factors (os, arch...)